packages/core-commands/src/site-editor-navigation-commands.js

```diff
@@ -373,33 +375,41 @@ const getSiteEditorBasicNavigationCommands = () =>
 					},
 				} );
 			}
-
-			result.push( {
-				name: 'core/edit-site/open-patterns',
-				label: __( 'Patterns' ),
-				icon: symbol,
-				callback: ( { close } ) => {
-					if ( canCreateTemplate ) {
-						if ( isSiteEditor ) {
-							history.navigate( '/pattern' );
+			if ( canCreateTemplate || canCreatePatterns ) {
```

Comment:
```suggestion
			if ( canCreatePatterns ) {
```

I think this conditional statement unintentionally exposes the command to users who have template creation permissions but not pattern creation permissions. Is there any reason to use `canCreateTemplate` here?

Reply:
Hey @t-hamano,

I got a little confused because of the callback where it handles both cases - if `canCreateTemplate` is present or not. This is the reason I added it in the outer if check.

But since you pointed it out, it makes sense to only have `canCreatePatterns` in the outer check as the command should only be visible to users who can actually create patterns. 

Should I update it to only use `canCreatePatterns`?

Reply:
> I got a little confused because of the callback where it handles both cases - if `canCreateTemplate` is present or not. 

I see, it may be a little difficult to understand.

In fact, the Patterns screen has two types:

- http://localhost:8888/wp-admin/edit.php?post_type=wp_block
- http://localhost:8888/wp-admin/site-editor.php?p=%2Fpattern

Since the latter is part of the site editor, we need to check not only `canCreatePatterns` but also `canCreateTemplate`. `canCreateTemplate` is generally used to check whether the current user can access the site editor. This is the reason that the `canCreateTemplate` exists within the callback function.

In summary, I think `canCreatePatterns` alone is sufficient for the outer check.

Reply:
Thanks for the clarification! That makes perfect sense now.